flaml/autogen/agent/user_proxy_agent.py

```diff
@@ -33,19 +36,61 @@ def __init__(
                     the number of auto reply reaches the max_consecutive_auto_reply.
                 (3) When "NEVER", the agent will never prompt for human input. Under this mode, the conversation stops
                     when the number of auto reply reaches the max_consecutive_auto_reply or when is_termination_msg is True.
+            function_map (dict[str, dict]): Mapping function names (passed to openai) to two types of functions:
+                    (1) A function to be called directly (dict): {
+                        "function" (Required, callable): a callable function that will be called
+                    }
+                    (2) A function in a class to be called (dict): {
+                        "class" (Required): an instance of a class.
+                        "func_name" (Optional, str): name of the function in the class. If not given the class will be called directly.
+                    }
+
+                Example 1:
+                def add_num_func(num_to_be_added):
```

Comment:
Use code blocks. Follow existing docstr format. Compile the doc and verify it looks good.